src/record.rs

```diff
@@ -106,15 +152,19 @@ impl Record {
         &self.header
     }
 
-    /// # Description
-    /// Serialize record to bytes
-    pub fn to_raw(&self) -> bincode::Result<Vec<u8>> {
+    pub(crate) fn to_partially_serialized(&self) -> Result<PartiallySerializedRecord> {
         let mut buf = self.header.to_raw()?;
-        trace!("raw header: len: {}", buf.len());
+        let header_len = buf.len();
         let raw_meta = self.meta.to_raw()?;
         buf.extend(&raw_meta);
         buf.extend(&self.data);
-        Ok(buf)
+        Ok(PartiallySerializedRecord::new(
+            buf,
+            header_len - 24,
+            header_len - 4,
```

Comment:
Add functions to `Header` that evaluate offsets:
`fn get_blob_offset_offset(header_len: usize) -> usize`
`fn get_data_checksum_offset(header_len: usize) -> usize`

These function can be used in `into_serialized_with_header_checksum`

Reply:
Also it is super critical to write tests, that verifies correctness of that functions. 
Test logic: serialize header, update data at the calculated offsets, deserialize header, verify that value was correctly updated

Reply:
Done